nemo/collections/asr/modules/transformer/transformer_modules.py

```diff
@@ -44,22 +44,37 @@ def __init__(self, hidden_size, max_sequence_length=512):
         self._max_sequence_length = max_sequence_length
         self._build_pos_enc(hidden_size=self._hidden_size, max_sequence_length=self._max_sequence_length)
 
-    def _build_pos_enc(self, hidden_size, max_sequence_length, device=None):
-        """
-        Builds/replaces pre-computed positional encoding.
-        """
-        pos_enc = torch.zeros(max_sequence_length, hidden_size, device=device)
-        position = torch.arange(0.0, max_sequence_length).unsqueeze(1)
+    @staticmethod
+    def _calc_pos_enc(hidden_size, sequence_length, start_position=0.0, device=None):
+        pos_enc = torch.zeros(sequence_length, hidden_size, device=device)
+        position = torch.arange(start_position, start_position + sequence_length).unsqueeze(1)
         coef = -math.log(10000.0) / hidden_size
         div_term = torch.exp(coef * torch.arange(0.0, hidden_size, 2))
         pos_enc[:, 0::2] = torch.sin(position * div_term)
         pos_enc[:, 1::2] = torch.cos(position * div_term)
         pos_enc.div_(math.sqrt(hidden_size))
+        return pos_enc
+
+    def _build_pos_enc(self, hidden_size, max_sequence_length, device=None):
+        """
+        Builds/replaces pre-computed positional encoding.
+        """
+        pos_enc = FixedPositionalEncoding._calc_pos_enc(hidden_size, max_sequence_length, device=None)
         self.register_buffer('pos_enc', pos_enc)
 
     def forward(self, position_ids):
-        embeddings = torch.embedding(self.pos_enc, position_ids)
-        return embeddings
+        """
+        Forward pass for pos_emb.
+        Checks if position_ids are in range.
+        If yes -> use lookup table
+        If no -> recalculate positions.
+        """
+        start_pos, seq_length = position_ids[0], position_ids.shape[1]
+        if start_pos + seq_length > self.max_sequence_length:
+            return FixedPositionalEncoding._calc_pos_enc(
+                self._hidden_size, seq_length, start_pos=start_pos, device=position_ids.device
```

Comment:
Fix codeQL error below. Also make sure pretrained loading works as is.

Reply:
This should have been removed from PR. See the other abs_emb pr and i'll delete this.

Reply:
resolved as this is removed from pr